kernel/src/snapshot.rs

```diff
@@ -422,11 +423,44 @@ mod tests {
     use object_store::path::Path;
     use object_store::ObjectStore;
 
+    use crate::actions::Format;
     use crate::engine::default::executor::tokio::TokioBackgroundExecutor;
     use crate::engine::default::filesystem::ObjectStoreFileSystemClient;
     use crate::engine::sync::SyncEngine;
     use crate::schema::StructType;
 
+    #[test]
+    fn test_snapshot_filtering() {
+        let path = std::fs::canonicalize(PathBuf::from("./tests/data/app-txn-checkpoint")).unwrap();
+        let url = url::Url::from_directory_path(path).unwrap();
+
+        let engine = SyncEngine::new();
+        let snapshot = Snapshot::try_new(url, &engine, Some(1)).unwrap();
```

Comment:
why version 1?

Reply:
Good point. This came from the code below.